### ISSUE TITLE
Bus error on sparc64

### ISSUE THREAD

**Issue by fsateler** [systemd/casync#60]
Tests are failing on sparc64 due to a bus error:

```
Processed casync/test-files
+ diff -q /var/tmp/test-casync.20172/original-seek.digest /var/tmp/test-casync.20172/extract-seek-catar.digest
+ diff -q /var/tmp/test-casync.20172/original-seek.digest /var/tmp/test-casync.20172/extract-seek-caidx.digest
+ diff -q /var/tmp/test-casync.20172/original-seek.digest /var/tmp/test-casync.20172/extract-seek-catar-direct.digest
+ diff -q /var/tmp/test-casync.20172/original-seek.digest /var/tmp/test-casync.20172/extract-seek-caidx-direct.digest
+ diff -q /var/tmp/test-casync.20172/original-seek.mtree /var/tmp/test-casync.20172/extract-seek-catar.mtree
+ diff -q /var/tmp/test-casync.20172/original-seek.mtree /var/tmp/test-casync.20172/extract-seek-caidx.mtree
+ CASYNC_SSH_PATH=/<<BUILDDIR>>/casync-1+65/test/pseudo-ssh
+ export CASYNC_SSH_PATH
+ CASYNC_REMOTE_PATH=/<<BUILDDIR>>/casync-1+65/obj-sparc64-linux-gnu/casync
+ export CASYNC_REMOTE_PATH
+ /<<BUILDDIR>>/casync-1+65/obj-sparc64-linux-gnu/casync -v --without=privileged --digest=sha256 list localhost:/var/tmp/test-casync.20172/test.caidx
Selected chunk sizes: min=16384..avg=65536..max=262144
-Using feature flags: nsec-time permissions symlinks fifos sockets flag-hidden flag-archive flag-noatime flag-compr flag-nocow flag-dirsync flag-sync flag-nocomp flag-projinherit xattrs
Excluding files with chattr(1) -d flag: yes
Excluding submounts: no
Digest algorithm: sha256
/<<BUILDDIR>>/casync-1+65/obj-sparc64-linux-gnu/test-script.sh: line 138: 58494 Bus error               "/<<BUILDDIR>>/casync-1+65/obj-sparc64-linux-gnu"/casync $PARAMS list localhost:$SCRATCH_DIR/test.caidx > $SCRATCH_DIR/test-remote.caidx.list
```

Full log here:
https://buildd.debian.org/status/fetch.php?pkg=casync&arch=sparc64&ver=1%2B65-1&stamp=1500566201&raw=0

(Yes, you can now `apt-get install casync` on your debian system :)

**Comment by keszybz** [comment]
Any chance you can provide the backtrace from that? Most likely it's some unaligned access...

**Comment by fsateler** [comment]
I'm not with my ssh key so I can't log into the sparc porterbox.  I'll do that when I get to my debian laptop.

**Comment by poettering** [comment]
hmm, i figure debian doesn't offer CI integration for upstream projects, does it? Would really love if we could integrate Debian's arch support somehow in our CI logic upstream. Would really love to see at least one BE arch hooked up, as they are exceedingly rare and major sources of portability pain... (and even better, an arch that throws SIGBUS on unaligned memory access)

**Comment by poettering** [comment]
Most likely fixed by #65. Please have a look!

**Comment by fsateler** [comment]
BTW, updating casync to include #65 does fix this issue:

https://buildd.debian.org/status/package.php?p=casync

We have all green now (at least on debian-supported archs).

> hmm, i figure debian doesn't offer CI integration for upstream projects, does it? Would really love if we could integrate Debian's arch support somehow in our CI logic upstream. Would really love to see at least one BE arch hooked up, as they are exceedingly rare and major sources of portability pain... (and even better, an arch that throws SIGBUS on unaligned memory access)

Sorry I somehow forgot about this. Unfortunately, we don't offer such CI integration. In fact, CI does not run on all archs even for debian packages. These machines serve only as build farms, not CI farms.
